metals/src/main/scala/scala/meta/internal/metals/NewFilesProvider.scala

```diff
@@ -23,28 +23,34 @@ class NewFilesProvider(
 ) {
 
   private val classPick = MetalsQuickPickItem(id = "class", label = "Class")
+  private val caseClassPick =
+    MetalsQuickPickItem(id = "case-class", label = "Case class")
   private val objectPick = MetalsQuickPickItem(id = "object", label = "Object")
   private val traitPick = MetalsQuickPickItem(id = "trait", label = "Trait")
   private val packageObjectPick =
     MetalsQuickPickItem(id = "package-object", label = "Package Object")
   private val worksheetPick =
     MetalsQuickPickItem(id = "worksheet", label = "Worksheet")
 
-  def createNewFileDialog(directoryUri: Option[URI]): Future[Unit] = {
+  def createNewFileDialog(
+      directoryUri: Option[URI],
+      name: Option[String]
```

Comment:
I've changed this to also accept a `name`, so that we can skip the name-picking dialog and just ask for a file kind.

This allows the code action to fully re-use the existing provider's flow, which I think is really nice